src/client.rs

```diff
@@ -57,7 +57,7 @@ impl Client {
     /// Sends a request to a client
     pub fn send_request(&self, request: &Request) -> Result<Response, Error> {
```

Comment:
Do you think I should also remove the `Clone` impl from `Request`?

Reply:
Sorry if it's obvious, but what's wrong with reusing the `Request`?

Reply:
Oh. It's probably because of the `id`. But I don't think `id` even has to be unique, does it? I'm looking at the https://www.jsonrpc.org/specification, and it says that the server has to reply with the same id, and that's all about it. This can be used to match the reply with the request, in case many were in flight at the same time, but I would imagine, most of the time this is not required since client waits for the reply.

On top of it:

```
        if response.id != request.id {                                                                                              
            return Err(Error::NonceMismatch);                                                                                       
        }   
```

looks like a footgun. As soon as one would start using `Client` concurrently, they would start getting errors, and not the right response.

Reply:
In case of http transport... I don't even think there's room for reordering... is there? From::

https://en.wikipedia.org/wiki/HTTP_pipelining

> as the limitation of HTTP 1.1 still applies: the server must send its responses in the same order that the requests were received—so the entire connection remains first-in-first-out

and in case of HTTP2, I would expect that it's http library that takes care of assembling responses in the right order.

So in JSONRPC over HTTP, the `id` makes sense only if one using the batching requests (https://www.jsonrpc.org/specification#batch), which doesn't seem to be implemented yet.

So if I'm not wrong, I guess the whole `id` management could be removed, unless:

* transports other than HTTP are supported
* support for batch send is added (in which case, request-uniqe id could be used, eg. to sort the response)

Reply:
Batch sending has been on the "when I find time" list for a while, I'd like to leave that door open.

But I think you're right that in the current implementation we could be using a constant id and it'd be fine.

Reply:
I guess the ids are mostly useful when using JSON-RPC over WebSockets.